spec/tasks/whitehall_importer_spec/create_image_revision_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+RSpec.describe Tasks::WhitehallImporter::CreateImageRevision do
+  include FixturesHelper
+
+  let(:img_url) { "https://assets.publishing.service.gov.uk/government/uploads/valid-image.jpg" }
+  let(:img_file) { "960x640.jpg" }
+  let(:whitehall_image) do
+    { "url": img_url, "caption": "Caption", "alt_text": "Alt text" }.with_indifferent_access
+  end
+
+  before do
+    stub_request(:get, img_url).to_return(
+      status: 200,
+      body: File.open(File.join(fixtures_path, "files", img_file), "rb").read,
+    )
+  end
+
+  context "A valid image is provided" do
+    before do
+      Tasks::WhitehallImporter::CreateImageRevision.call(whitehall_image)
+    end
+
+    it "should create an Image::MetadataRevision that preserves the alt text and caption" do
+      expect(Image::MetadataRevision.count).to eq(1)
+      expect(Image::MetadataRevision.first.caption).to eq(whitehall_image["caption"])
+      expect(Image::MetadataRevision.first.alt_text).to eq(whitehall_image["alt_text"])
+    end
+
+    it "should create an Image::BlobRevision that preserves the filename" do
+      expect(Image::BlobRevision.count).to eq(1)
+      expect(Image::BlobRevision.first.filename).to eq("valid-image.jpg")
+    end
+
+    it "should create an Image::Revision that references the Metadata and Blob revisions" do
+      expect(Image::Revision.count).to eq(1)
+      expect(Image::Revision.first.blob_revision).to eq(Image::BlobRevision.first)
+      expect(Image::Revision.first.metadata_revision).to eq(Image::MetadataRevision.first)
+    end
+  end
+
+  context "Image not available" do
+    before do
+      stub_request(:get, img_url).to_return(status: 404)
+    end
+
+    it "should raise an AbortImageError" do
+      expect { Tasks::WhitehallImporter::CreateImageRevision.call(whitehall_image) }.to raise_error(
+        AbortImageError,
+        "Image does not exist: #{img_url}",
+      )
+    end
+
+    it "should not create any revision instances" do
+      expect(Image::MetadataRevision.count).to eq(0)
+      expect(Image::BlobRevision.count).to eq(0)
+      expect(Image::Revision.count).to eq(0)
+    end
+  end
+
+  context "Image is wrong type" do
+    let(:img_url) { "https://assets.publishing.service.gov.uk/government/uploads/vector.svg" }
+    let(:img_file) { "coffee.svg" }
+
+    it "should pass through ImageUploadChecker and raise an AbortImageError" do
+      expect(Requirements::ImageUploadChecker).to receive(:new).and_call_original
+      expect { Tasks::WhitehallImporter::CreateImageRevision.call(whitehall_image) }.to raise_error(
+        AbortImageError,
+        "Select a jpg, png or gif image file",
```

Comment:
It'd be more consistent with the existing test setup to use: `I18n.t!("requirements.image_upload.unsupported_type.form_message")` as we have a consistency for not using strings from the locale file directly.